aten/src/ATen/native/cpu/UnaryOpsKernel.cpp

```diff
@@ -32,99 +34,96 @@ static void parallel_apply(Tensor& result, const Tensor& self, F f) {
 
   static tbb::affinity_partitioner ap;
 
-  auto arr_out = result.data<scalar_t>();
-  auto arr_in = self.data<scalar_t>();
-  int64_t size = self.numel();
-  if (size < internal::TBB_GRAIN_SIZE) {
-    unary_kernel(arr_out, arr_in, size, f);
+  if (result.is_contiguous() && self.is_contiguous()) {
+    auto arr_out = result.data<scalar_t>();
+    auto arr_in = self.data<scalar_t>();
+    int64_t size = self.numel();
+    if (size < internal::TBB_GRAIN_SIZE) {
+      unary_kernel(arr_out, arr_in, size, f);
+    } else {
+      tbb::parallel_for(
+          tbb::blocked_range<int64_t>(0, size, internal::TBB_GRAIN_SIZE),
+          [&](const tbb::blocked_range<int64_t>& r) {
+            auto size = r.end() - r.begin();
+            unary_kernel(arr_out + r.begin(), arr_in + r.begin(), size, f);
+          },
+          ap);
+    }
   } else {
-    tbb::parallel_for(
-        tbb::blocked_range<int64_t>(0, size, internal::TBB_GRAIN_SIZE),
-        [&](const tbb::blocked_range<int64_t>& r) {
-          auto size = r.end() - r.begin();
-          unary_kernel(arr_out + r.begin(), arr_in + r.begin(), size, f);
-        },
-        ap);
+    CPU_tensor_apply2<scalar_t, scalar_t>(
+        result, self, [f](scalar_t& a_val, const scalar_t& b_val) {
+          unary_kernel(&a_val, &b_val, 1, f);
```

Comment:
Is it the plan to start finding smaller contiguous sub-regions in a tensor, and still use vectorization over them?